test/bidirectional_graph_test.rb

```diff
@@ -0,0 +1,166 @@
+require 'test_helper'
+
+require 'rgl/adjacency'
+require 'rgl/bidirectional'
+
+include RGL
+include RGL::Edge
+
+class TestBidirectionalGraph < Test::Unit::TestCase
```

Comment:
We could in addition reuse `TestDirectedGraph` here only overriding setup method:

```suggestion
class TestDirectedGraphBidirected < TestDirectedGraph
  def setup
    @dg = BidirectionalAdjacencyGraph.new
    # TODO: refactor this in a protected setup function which we reuse here
    [[1, 2], [2, 3], [3, 2], [2, 4]].each do |(src, target)|
      @dg.add_edge(src, target)
    end
  end
end
```

All tests in the super test class should then pass.